doc/sphinx/language/gallina-extensions.rst

```diff
@@ -1,2152 +1,2356 @@
-
-\begin{coq_example}
-Set Printing Projections.
-Check top half.
-\end{coq_example}
-
-The corresponding grammar rules are given in Figure~\ref{fig:projsyntax}.
-When {\qualid} denotes a projection, the syntax {\tt
-  {\term}.({\qualid})} is equivalent to {\qualid~\term}, the syntax
-{\term}{\tt .(}{\qualid}~{\termarg}$_1$ {\ldots} {\termarg}$_n${\tt )} to
-{\qualid~{\termarg}$_1$ {\ldots} {\termarg}$_n$~\term}, and the syntax
-{\term}{\tt .(@}{\qualid}~{\term}$_1$~\ldots~{\term}$_n${\tt )} to
-{@\qualid~{\term}$_1$ {\ldots} {\term}$_n$~\term}. In each case, {\term}
-is the object projected and the other arguments are the parameters of
-the inductive type.
-
-\begin{figure}[t]
-\begin{centerframe}
-\begin{tabular}{lcl}
-{\term} & ++= & {\term} {\tt .(} {\qualid} {\tt )}\\
- & $|$ & {\term} {\tt .(} {\qualid} \nelist{\termarg}{} {\tt )}\\
- & $|$ & {\term} {\tt .(} {@}{\qualid} \nelist{\term}{} {\tt )}
-\end{tabular}
-\end{centerframe}
-\caption{Syntax for \texttt{Record} projections}
-\label{fig:projsyntax}
-\end{figure}
-
-\begin{coq_eval}
-Reset Initial.
-\end{coq_eval}
-
-\begin{Remarks}
-
-\item Records defined with the {\tt Record} keyword are not allowed to be
-recursive (references to the record's name in the type of its field
-raises an  error). To define recursive records, one can use the {\tt
-Inductive} and {\tt CoInductive} keywords, resulting in an inductive
-or co-inductive record.
-A \emph{caveat}, however, is that records
-cannot appear in mutually inductive (or co-inductive) definitions.
-
-\item Induction schemes are automatically generated for inductive records.
-Automatic generation of induction schemes for non-recursive records
-defined with the {\tt Record} keyword can be activated with the
-{\tt Nonrecursive Elimination Schemes} option
-(see~\ref{set-nonrecursive-elimination-schemes}).
-
-\item {\tt Structure} is a synonym of the keyword {\tt Record}.
-
-\end{Remarks}
-
-\begin{Warnings}
-\item {\tt {\ident$_i$} cannot be defined.}
+.. coqtop:: all
+
+   Eval compute in half.(top).
+
+It can be activated for printing with
+
+.. cmd:: Set Printing Projections.
+
+.. example::
+
+    .. coqtop:: all
+
+       Set Printing Projections.
+       Check top half.
+
+.. _record_projections_grammar:
+
+  .. productionlist:: terms
+     term : term `.` ( qualid )
+          : | term `.` ( qualid arg … arg )
+          : | term `.` ( @`qualid` `term` … `term` )
+
+  Syntax of Record projections
+
+The corresponding grammar rules are given in the preceding grammar. When `qualid`
+denotes a projection, the syntax `term.(qualid)` is equivalent to `qualid term`,
+the syntax `term.(qualid` |arg_1| |arg_n| `)` to `qualid` |arg_1| `…` |arg_n| `term`,
+and the syntax `term.(@qualid` |term_1| |term_n| `)` to `@qualid` |term_1| `…` |term_n| `term`.
+In each case, `term` is the object projected and the
+other arguments are the parameters of the inductive type.
+
+.. note::. Records defined with the ``Record`` keyword are not allowed to be
```

Comment:
What is this `.` after the two `::`?

Reply:
Sphinx is interpreting this as a comment and not displaying it in the rendered HTML!